### PR TITLE
Test split

### PR DESCRIPTION
**Pull request by rustyrussell**
Split the test.sh script for easier maintenance, clarity, and speed.  Changing them also uncovered two new bugs, which I fixed first.

### CODE REVIEW

File: daemon/test/scripts/helpers.sh [hunk 1]
```diff
@@ -188,6 +188,14 @@ start_lightningd()
 
     PORT2=`get_info_field "$LCLI2" port`
     [ $NUM_LIGHTNINGD = 2 ] || PORT3=`get_info_field "$LCLI3" port`
+
+    # Make a payment into a P2SH for anchor.
+    P2SHADDR=`$LCLI1 newaddr | sed -n 's/{ "address" : "\(.*\)" }/\1/p'`
+    FUND_INPUT_TXID=`$CLI sendtoaddress $P2SHADDR 0.01`
+    FUND_INPUT_TX=`$CLI getrawtransaction $FUND_INPUT_TXID`
```
Comment (cdecker) [inline daemon/test/scripts/helpers.sh]: Defining a variable in a function that is then used outside a function is a bit confusing. Can't we return it?

**Comment by cdecker** [comment]
tACK :+1:

File: daemon/test/scripts/helpers.sh (hunk 1 above)
Comment (rustyrussell) [inline daemon/test/scripts/helpers.sh]: Agreed, global vars are unnatural.  This method is a result of history, not design.  I'll append a cleanup patch.

Comment (rustyrussell) [inline daemon/test/scripts/helpers.sh]: Erk, no.  In shell you return things via stdout, and there is other output here.  Maybe one day we'll rewrite in python, though that's probably overkill...

**Pull request merged**